### DISCUSSION TITLE
Add Sort Transformation

### DISCUSSION THREAD

**Discussion by adeverteuil** [grafana/grafana#38174] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

A Transformation to sort series by legend, numeric value, or label value.

**Why is this needed**:

Prometheus can sort by value, but not by label value. The ordering of series sometimes appears random and changes at every reload. The Stat, Gauge and Bar Gauge panels display series in the order received from the data source. A sorting transformation is needed to present the data in a more readable way.

![screenshot 2020-11-09 102001](https://user-images.githubusercontent.com/3270755/98559893-2b6cca00-2275-11eb-9d47-dd90b8ab3472.png)

**See also**:

* This closed feature request in prometheus/prometheus. This won't be implemented in the Prometheus API.
   https://github.com/prometheus/prometheus/issues/414
* Similar feature request in grafana/grafana. I'm reframing this as a feature request for a new Transformation:
   https://github.com/grafana/grafana/issues/17245

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424).

I think this has been implemented - please see https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/#sort-by

Feel free to make an [issue](https://github.com/grafana/grafana/issues) if this is not a solution and you would like to request a feature.
